src/Client.php

```diff
@@ -99,6 +99,14 @@ public function __construct($uri = 'mongodb://127.0.0.1/', array $uriOptions = [
             throw InvalidArgumentException::invalidType('"typeMap" driver option', $driverOptions['typeMap'], 'array');
         }
 
+        if (isset($driverOptions['autoEncryption']['keyVaultClient'])) {
+            if ($driverOptions['autoEncryption']['keyVaultClient'] instanceof self) {
+                $driverOptions['autoEncryption']['keyVaultClient'] = $driverOptions['autoEncryption']['keyVaultClient']->manager;
+            } elseif (! $driverOptions['autoEncryption']['keyVaultClient'] instanceof Manager) {
+                throw InvalidArgumentException::invalidType('"keyVaultClient" autoEncryption option', $driverOptions['autoEncryption']['keyVaultClient'], [self::class, Manager::class]);
```

Comment:
Note that `functions.php` already happened to pass "array or object" for this sort of error. I'll defer to you if you'd like to keep the new code in `invalidType()` (kudos for handling the case for 3+ types :smile:) or replace this with a string.

Reply:
Since we're dealing with class names, the string concatenation when calling `invalidType` wouldn't have looked nice.
(I've been reminded of proper Oxford comma usage often enough to remember it now, hence the special case for 3+ types)